packages/web-config-server/src/apiV1/measureData.js

```diff
@@ -152,69 +196,13 @@ export default class extends DataAggregatingRouteHandler {
         }
       }),
     );
-
-    // start fetching options
-    const optionsTasks = overlays.map(o => this.fetchMeasureOptions(o, this.query));
     // start fetching actual data
     const shouldFetchSiblings = this.query.shouldShowAllParentCountryResults === 'true';
-    const dataTasks = overlays.map(o => this.fetchMeasureData(o, shouldFetchSiblings));
-
-    // wait for fetches to complete
-    const measureOptions = await Promise.all(optionsTasks);
-    const measureDataResponsesByMeasureId = (
-      await Promise.all(dataTasks)
-    ).reduce((dataResponse, current) => ({ ...dataResponse, ...current }));
-
-    /* Data arrives as an array of responses (one for each measure) containing an array of org
-     * units. We need to rearrange it so that it's a 1D array of objects with the values
-     * assigned to the appropriate keys, and replace the keys with the ids of the overlay
-     * they came from (this is to avoid duplicate 'value' keys which causes a bug)
-     *
-     * measureDataResponsesByMeasureId: {
-     *   measure_id_1: {
-     *     data: [
-     *       { organisationUnitCode: 'OrgA', measureZ: 0 },
-     *       { organisationUnitCode: 'OrgB', measureZ: 1 },
-     *     ],
-     *     period: {
-     *       latestAvailable: '20200301',
-     *       earliestAvailable: '20190501',
-     *       requested: '201901;201902;201903...',
-     *     }
-     *   }
-     *   measure_id_2: {
-     *     data: [
-     *       { organisationUnitCode: 'OrgA', measureY: 100 },
-     *       { organisationUnitCode: 'OrgB', measureY: -100 },
-     *     ],
-     *     period: {
-     *       latestAvailable: '20200401',
-     *       earliestAvailable: '20190501',
-     *       requested: '201901;201902;201903...',
-     *     }
-     *   }
-     * }
-     *
-     * measureData: [
-     *  { organisationUnitCode: 'OrgA', measure_id_2: 100, measure_id_1: 0 },
-     *  { organisationUnitCode: 'OrgB', measure_id_2: -100, measure_id_1: 1 },
-     * ]
-     * period: {
-     *   latestAvailable: '20200401',
-     *   earliestAvailable: '20190501',
-     *   requested: '201901;201902;201903...',
-     * }
-     */
-
-    const { period, data: measureData } = buildMeasureData(
-      overlays,
-      measureDataResponsesByMeasureId,
+    const responseData = await Promise.all(
+      overlays.map(o => this.fetchMeasureData(o, shouldFetchSiblings)),
     );
-
-    measureOptions
-      .filter(mo => mo.displayedValueKey)
-      .filter(mo => !mo.disableRenameLegend)
-      .map(mo => updateLegendFromDisplayedValueKey(mo, measureData));
```

Comment:
Group these code as function `fetchMeasureOptions` (line 220 - 225), and rename existed function `fetchMeasureOptions` to `fetchMeasureOption`